FlowCrypt/Functionality/Services/Contacts Services/LocalContactsProvider.swift

```diff
@@ -61,21 +58,37 @@ extension LocalContactsProvider: LocalContactsProviderType {
     }
 
     func searchContact(with email: String) -> Contact? {
+        guard let contactObject = find(with: email) else { return nil }
+        return Contact(contactObject)
+    }
+
+    func getAllContacts() -> [Contact] {
         localContactsCache.realm
             .objects(ContactObject.self)
-            .first(where: { $0.email == email })
-            .map { Contact($0) }
+            .map { object in
+                let keyDetails = object.pubKeys
+                                    .compactMap { try? core.parseKeys(armoredOrBinary: $0.key.data()).keyDetails }
+                                    .flatMap { $0 }
+                return Contact(object, keyDetails: Array(keyDetails))
+            }
+            .sorted(by: { $0.email > $1.email })
     }
 
-    func getAllContacts() -> [Contact] {
-        Array(
-            localContactsCache.realm
-                .objects(ContactObject.self)
-                .map {
-                    let keyDetail = try? core.parseKeys(armoredOrBinary: $0.pubKey.data()).keyDetails.first
-                    return Contact($0, keyDetail: keyDetail)
+    func remove(pubKey: String, for email: String) {
```

Comment:
This needs a different signature - removing by email+fingerprint of the key, instead of the key string.

Reply:
for me - Also need to double check the structure of the table later, and indexes